diff --git a/AltStore/Components/AppCardCollectionViewCell.swift b/AltStore/Components/AppCardCollectionViewCell.swift
index 68c5320..156b96f 100644
--- a/AltStore/Components/AppCardCollectionViewCell.swift
+++ b/AltStore/Components/AppCardCollectionViewCell.swift
@@ -187,24 +187,8 @@ private extension AppCardCollectionViewCell
             let itemSize = NSCollectionLayoutSize(widthDimension: .estimated(1), heightDimension: .fractionalHeight(1.0))
             let item = NSCollectionLayoutItem(layoutSize: itemSize)
             
-            if numberOfVisibleScreenshots == 1
-            {
-                // If there's only one screenshot visible initially, we'll (reluctantly) opt-in to flexible spacing on both sides.
-                // This ensures the items are always centered, but may result in larger spacings between items than we'd prefer.
-                item.edgeSpacing = NSCollectionLayoutEdgeSpacing(leading: .flexible(0), top: nil, trailing: .flexible(0), bottom: nil)
-            }
-            else
-            {
-                // Otherwise, only have flexible spacing on the leading edge, which will be balanced by trailingGroup's flexible trailing spacing.
-                item.edgeSpacing = NSCollectionLayoutEdgeSpacing(leading: .flexible(0), top: nil, trailing: nil, bottom: nil)
-            }
-            
-            let groupItem = NSCollectionLayoutItem(layoutSize: itemSize)
-            let trailingGroup = NSCollectionLayoutGroup.horizontal(layoutSize: itemSize, subitems: [groupItem])
-            trailingGroup.edgeSpacing = NSCollectionLayoutEdgeSpacing(leading: nil, top: nil, trailing: .flexible(0), bottom: nil)
-            
             let groupSize = NSCollectionLayoutSize(widthDimension: .fractionalWidth(1.0), heightDimension: .fractionalHeight(1.0))
-            let group = NSCollectionLayoutGroup.horizontal(layoutSize: groupSize, subitems: [item, trailingGroup])
+            let group = NSCollectionLayoutGroup.horizontal(layoutSize: groupSize, subitems: [item])
             group.interItemSpacing = .fixed(minimumItemSpacing)
             
             if numberOfVisibleScreenshots < self.screenshots.count
